dxr/app.py

```diff
@@ -162,6 +164,27 @@ def _tree_tuples(trees, tree, query_text, is_case_sensitive):
              description)
             for t, description in trees.iteritems()]
 
+@dxr_blueprint.route('/<tree>/raw/<path:path>')
+def send_raw(tree, path):
+    """Send raw data at path from tree, for binary things like images."""
+    query = {
+            'filter': {
+                'term': {'path': path}
+                }
+            }
+    index = current_app.config['ES_ALIASES'][tree]
+    results = current_app.es.search(
+            query,
+            index=index,
+            doc_type=FILE, size=1)
+    try:
+        # we explicitly get index 0 because there should be exactly 1 result
+        data = results['hits']['hits'][0]['_source']['raw_data'][0]
+        dataFile = StringIO(data.decode("base64"))
+        return send_file(dataFile, mimetype=guess_type(path)[0])
+    except IndexError: # couldn't find the image
+        raise IndexError("Couldn't find image for %s.\nSearch: %s\nResults: %s"
```

Comment:
The end user shouldn't be able to get DXR to throw exceptions by passing in bad data. So let's make this a 404 instead.